enhancements/windows-containers/windows-node-egress-proxy.md

```diff
@@ -317,11 +285,57 @@ to WMCO 9.0.0, the proxy support will be available.
+* There is another possible way for WMCO to [retrieve the proxy variables](#configuring-proxy-environment-variables). WMCO
+  can read the environment variables from its own pod spec (vars are injected + updated by OLM). The difficulty of this
+  approach comes from figuring out when we need to update node's env vars. Ideally such reconfiguring happens only when
+  the values change in the pod spec, but how can we detect if the proxy env vars changed or the operator just restarted 
+  for some other reason? We want to avoid kicking off reconfigurations of all nodes every time the operator restarts.
+* In order to avoid a system reboot after setting node environment variables, we can reconcile services by setting their
+  environment variables, and then restarting the services. These changes are ephemeral, though, if the instance reboots.
+  ```powershell
+  [string[]] $envVars = @("HTTP_PROXY=http://<username>:<pswd>@<ip>:<port>", "NO_PROXY=123.example.com,10.88.0.0/16")
+  Set-ItemProperty HKLM:SYSTEM\CurrentControlSet\Services\<$SERVICE_NAME> -Name Environment -Value $envVars
+  Restart-Service <$SERVICE_NAME>
+  ```
+* Also note that there is another way to get the [trusted CA data](#configuring-custom-trusted-certififactes) required
+  rather than accessing the ConfigMap directly, but I prefer not to follow it. It leaves open the same concern about
+  unnecessary reconciliations -- how to detect if the operator restart was due to a trust bundle file change or the pod
+  just restarted for another reason? For completeness, I will list it out the apporach here.
+  * Update the operator’s Deployment to support trusted CA injection by mounting the trusted CA ConfigMap
+  ```yaml
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: windows-machine-config-operator
+    namespace: openshift-windows-machine-config-operator
+    annotations:
+      config.openshift.io/inject-proxy: windows-machine-config-operator
+  spec:
+      ...
+        containers:
+          - name: windows-machine-config-operator
+            volumeMounts:
+            - name: trusted-ca
+              mountPath: /etc/pki/ca-trust/extracted/pem
+              readOnly: true
+        - name: trusted-ca
+          configMap:
+            name: trusted-ca
+            items:
+              - key: ca-bundle.crt
+                path: tls-ca-bundle.pem
+  ...
+  ```
+  * Create a file watcher that watches changes to the mounted trust bundle that kills the main operator process and
+    allows the backing k8s Deployment to start a new Pod that mounts the updated trust bundle. 
+    Implementation example: [cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator/pull/334)
+* Instead of adding a new vSphere job, we can leverage an [existing proxy test workflow on AWS](https://steps.ci.openshift.org/workflow/openshift-e2e-aws-proxy).
```

Comment:
Please preface this with a testing alternative sub-heading as this is not a design alternative.